x/authenticator/authenticator/message_filter.go

```diff
@@ -175,57 +237,3 @@ func IsJsonSuperset(a, b []byte) error {
 
 	return isSuperset(av, bv)
 }
-
-type SimplifiedMsg struct {
-	MsgType string          `json:"type"`
-	Value   json.RawMessage `json:"value"`
-}
-
-func (m MessageFilterAuthenticator) Authenticate(ctx sdk.Context, request iface.AuthenticationRequest) iface.AuthenticationResult {
-	encodedMsg, err := json.Marshal(SimplifiedMsg{
-		MsgType: request.Msg.TypeURL,
-		Value:   request.Msg.Value,
-	})
-
-	if err != nil {
-		return iface.NotAuthenticated()
-	}
-	// Check that the encoding is a superset of the pattern
-	err = IsJsonSuperset(m.pattern, encodedMsg)
-	if err != nil {
-		// emit an event with the error
-		ctx.EventManager().EmitEvent(
-			sdk.NewEvent(
-				"AuthenticatorError",
-				sdk.NewAttribute("error", err.Error()),
-			),
-		)
-
-		return iface.NotAuthenticated()
-	}
-	return iface.Authenticated()
-}
-
-func (m MessageFilterAuthenticator) ConfirmExecution(ctx sdk.Context, request iface.AuthenticationRequest) iface.ConfirmationResult {
-	return iface.Confirm()
-}
-
-func (m MessageFilterAuthenticator) OnAuthenticatorAdded(ctx sdk.Context, account sdk.AccAddress, data []byte) error {
```

Comment:
We should keep the validation, right? If we can't serialize the value, adding the authenticator should fail.

Reply:
this is still there, jsut moved to the top of the file :smile: